db/models.py

```diff
@@ -50,3 +55,49 @@
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return f"Order: {self.created_at}"
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(to=MovieSession, on_delete=models.CASCADE)
+    order = models.ForeignKey(to=Order, on_delete=models.CASCADE)
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["movie_session", "seat", "row"],
+                name="unique_ticket",
+            )
+        ]
+
+    def clean(self):
+        if self.row <= 0 and self.row > self.movie_session.cinema_hall.rows:
+            raise ValidationError({"row": ["Row number must be in available range."]})
+        if self.seat <= 0 and self.seat > self.movie_session.cinema_hall.rows:
+            raise ValidationError({"seat": ["Seat number must be in available range."]})
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
+        super.save(*args, **kwargs)
```

Comment:
There is a typo in `super.save(*args, **kwargs)`. It should be `super().save(*args, **kwargs)` to correctly call the parent class's save method.